exporter/splunkhecexporter/client.go

```diff
@@ -247,8 +257,7 @@ func (c *client) fillMetricsBuffer(metrics pmetric.Metrics, buf buffer, is iterS
 	var permanentErrors []error
 	jsonStream := jsonStreamPool.Get().(*jsoniter.Stream)
 	defer jsonStreamPool.Put(jsonStream)
-
-	tempBuf := bytes.NewBuffer(make([]byte, 0, c.config.MaxContentLengthMetrics))
+	jsonStream.Reset(buf)
```

Comment:
Same here. `uint(jsonStream.Buffered()) > c.config.MaxEventSize` condition is ignored